### PR TITLE
Update remote Ingress on annotation change

### PR DESCRIPTION
**Pull request by ytsarev**
* shouldUpdate function which detects the need
  of remoteIngress update and also mutates Spec and
  annotation of associated remoteIngressCopy
* Update remoteIngress according to result of
  shouldUpdate function
* Unit tests
* Annotation e2e test

### CODE REVIEW

File: pkg/controllers/follow/ingress/controller_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: "test"},
+		},
+	}
+	ingress.Annotations = make(map[string]string)
+	ingress.Annotations = map[string]string{"annotate": "me"}
+	remoteIngress := v1beta1.Ingress{
+		Spec: v1beta1.IngressSpec{
+			Backend: &v1beta1.IngressBackend{
+				ServiceName: "test2"},
```
Comment (adrienjt) [inline pkg/controllers/follow/ingress/controller_test.go:57]: Shouldn't you use identical specs in this test?

File: pkg/controllers/follow/ingress/controller_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		},
+	}
+	remoteIngress.Annotations = make(map[string]string)
+
+	updatedIngress, shouldUpdate := r.shouldUpdate(&remoteIngress, &ingress)
+	if shouldUpdate != true {
+		t.Error("Ingresses with different annotations should trigger update")
+	}
+
+	assert.Equal(t, updatedIngress.Annotations, ingress.Annotations)
```
Comment (adrienjt) [inline pkg/controllers/follow/ingress/controller_test.go:67]: Could you add another test where both Ingresses have (mismatching) Admiralty-prefixed annotations (that aren't supposed to be updated), esp. `is-delegate`, the cross-cluster owner refs, and I may forgot what else? You could get real key/value pairs from an e2e test for the unit test to be representative, and to help future developers understand why it's important.

**Review by adrienjt — CHANGES_REQUESTED** [review]
Thank you very much for submitting this PR.

First of all, could you please sign-off your commit: https://github.com/admiraltyio/admiralty/blob/master/CONTRIBUTING.md#dco-sign-off

I have a couple of comments regarding unit tests (see line-attached comments).

If time allows, could you also add an ingress annotation update to `test/e2e/ingress/test.sh` and verify that it works? If not, at least add a TODO comment there. (I realize that it might be easier when e2e tests are ported from Bash to Go.)

File: pkg/controllers/follow/ingress/controller_test.go (hunk 1 above)
Comment (ytsarev) [reply pkg/controllers/follow/ingress/controller_test.go:57]: good catch

File: pkg/controllers/follow/ingress/controller_test.go (hunk 2 above)
Comment (ytsarev) [reply pkg/controllers/follow/ingress/controller_test.go:67]: @adrienjt added to tests (positive/negative) with the real set of annotations that I took from runtime

**Comment by ytsarev** [comment]
@adrienjt I believe I addressed all the issues, please check it out

**Review by adrienjt — APPROVED** [review]

**Pull request merged**